src/placeos-driver/storage/edge-storage.cr

```diff
@@ -2,41 +2,38 @@ require "../storage"
 require "../protocol"
 
 class PlaceOS::Driver::EdgeStorage < PlaceOS::Driver::Storage
-  def initialize(@id : String, @prefix = DEFAULT_PREFIX)
-    super()
-    @hash_key = "#{prefix}/#{@id}"
+  private getter hash : Hash(String, String) = {} of String => String
+  forward_missing_to hash
+
+  # This is the same as setting a value as this is often used when
+  # a hash value is updated and we want to notify of this change.
+  def signal_status(status_name) : String?
+    status_name = status_name.to_s
+    json_value = self[status_name]?
+    adjusted_value = json_value || "null"
+    PlaceOS::Driver::Protocol.instance.request(hash_key, "hset", "#{status_name}\x03#{adjusted_value}", raw: true)
+    json_value
   end
 
-  getter hash_key : String
-  getter id : String
-  getter prefix : String
+  # Hash methods
+  #################################################################################################
 
   def []=(status_name, json_value)
     status_name = status_name.to_s
     adjusted_value = json_value.to_s.presence
```

Comment:
Unchanged here, but worth flagging for discussion...

Is there a reason for forcing blank strings to `nil`? These are two different things and semantics of this is potentially useful for driver authors.

Reply:
@stakach

Reply:
@kimburgess want to spin this into an issue?

Reply:
https://github.com/PlaceOS/driver/issues/67